### PR TITLE
Add support for range iterator

### PR DESCRIPTION
**Pull request by jerryzhenleicai**
Now it's possible to iterate over a subset of keys (range of keys):

    iter = LevelDB.db_range_iterator(db, [0xb, 0xa], [0xc, 0xd])
    for (k, v) in iter:
        ...

### CODE REVIEW

**Comment by gdkrmr** [comment]
Cool! How does the ordering of keys work?

**Comment by jerryzhenleicai** [comment]
> Cool! How does the ordering of keys work?

Not 100% sure about your question Guido. LevelDB's keys are inherently ordered. So we get it for free

File: README.md [hunk 1]
```diff
@@ -86,6 +86,15 @@ julia> for (key, value) in db
 ```
 Iterate over all `key => value` pairs in a `LevelDB.DB`.
 
+
+```julia
+julia> for (key, value) in db_range_iterator(db, key1, key2)
```
[the remark this answers is not in the document]
Comment (jerryzhenleicai) [reply README.md]: Good idea. I made the corresponding modifications in the new commit.

**Comment by gdkrmr** [comment]
Looking at the leveldb source code, you can give your own comparator, but the standard one is "lexicographic byte-wise", i.e. like you would order words in a dictionary, except that you use bytes:

https://github.com/google/leveldb/blob/c784d63b931d07895833fb80185b10d44ad63cce/include/leveldb/comparator.h#L57

**Comment by gdkrmr** [comment]
thanks a lot!

**Pull request merged**